fn/io.go

```diff
@@ -0,0 +1,53 @@
+package fn
+
+import (
+	"fmt"
+	"os"
+)
+
+// FileRemovalOnErr is a flag used to indicate the removal or not of a file on a
+// failure.
+type FileRemovalOnErr bool
+
+const (
+	// KeepFileOnErr indicates no deletion of the file on error.
+	KeepFileOnErr FileRemovalOnErr = false
+
+	// RemoveFileOnErr indicates deletion of the file on error.
+	RemoveFileOnErr FileRemovalOnErr = true
+)
+
+// WriteFile synchronous writes data to the named file.
+// If the file does not exist, WriteFile creates it with permissions perm
+// (before umask); otherwise WriteFile truncates it before writing, without
+// changing permissions.
+// By opening the file with O_SYNC, its ensures the data is written to disk.
+// The removeOnErr flag controls the removal or not of the file on error.
+func WriteFile(name string, data []byte, perm os.FileMode,
+	removeOnErr FileRemovalOnErr) (err error) {
+
+	defer func() {
+		if err != nil && removeOnErr == RemoveFileOnErr {
+			fmt.Printf("err > %s\n", err)
```

Comment:
We shouldn't directly print anything. I think it's enough that we return the error.

Reply:
Print line removed